src/CBT.UnifiedOutputDir/build/CBT.UnifiedOutputDir.props

```diff
@@ -21,6 +21,12 @@
     <CBTIntermediateOutputPathPartial>$(MSBuildProjectFullPath.SubString($(EnlistmentRoot.Length)))</CBTIntermediateOutputPathPartial>
     <IntermediateOutputPath Condition="'$(CBTEnableHashedIntermediateOutputPath)' == 'true'">$([System.IO.Path]::Combine($(IntermediateOutputPath),$(CBTIntermediateOutputPathPartial.Length.ToString())))\</IntermediateOutputPath>
 
+    <!-- Ensure nuGet restore places generated props correctly. -->
+    <MSBuildProjectExtensionsPath>$(IntermediateOutputPath)\</MSBuildProjectExtensionsPath>
+    <RestoreOutputPath>$(MSBuildProjectExtensionsPath)</RestoreOutputPath>
+    <!-- In the projectJson package restore scenario you can not redefine the location of the json file.  If the ProjectAssetsFile is set then it will be assumed to be a PackageReference project.  -->
+    <ProjectAssetsFile Condition="!Exists('$(MSBuildProjectDirectory)\project.json') And !Exists('$(MSBuildProjectDirectory)\package.config')" >$(MSBuildProjectExtensionsPath)project.assets.json</ProjectAssetsFile>
```

Comment:
I conditioned this ProjectAssetsFile since it only pertains to the ProjectReference scenario.  Perhaps there is a better way to do this but by the time this props is imported PackageReference may not be known and by the time the targets is loaded it is to late.  Add to this the evaluation hell nuget does and I went with this existence check.  At least for now.